### DISCUSSION TITLE
Can I remove all profile related features when embedding Tabby?

### DISCUSSION THREAD

**Discussion by yume-chan** [Eugeny/tabby#8282] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/1330321/233766798-a427d7d9-b7a7-4522-a131-58862c60b20a.png)

I'm embedding Tabby into my project using NPM packages. Since I only have one profile, the "Profiles & connections" button, the profile selector and the profiles setting page are not useful to me.

Is it possible to remove them only leaving the button I added to create a new tab directly?

I think I need to remove the `CoreCommandProvider` and `ProfilesSettingsTabProvider` injections, but it's not possible without modifying the source code and rebuild.

**Comment by Eugeny** [comment, marked as answer]
I've just added config options that will let you blacklist individual commands and providers - in your case, this should do the trick:

```
providerBlacklist: 
- settings:ProfilesSettingsTabProvider
commandBlacklist: 
- core:profile-selector
```

I've also built and pushed new packages to NPM (`1.0.197-nightly.0`)

**Reply by yume-chan** [reply]
Thank you! `commandBlacklist` was easy to add, but `providerBlacklist` took me some time, as `tabby-web-container` didn't set `window['pluginModules']` like the Electron app does, I have to copy this part of code to convert them into correct shape.

https://github.com/Eugeny/tabby/blob/194342d6fa4f6c44b3f10baed39d2350f5fb9ef2/web/entry.ts#L28-L37

**Reply by yume-chan** [reply]
Some other minor issues:

1. Published packages have outdated peer dependencies, causes warnings from package managers, should it be `">= 12.0.0"`?
    https://github.com/Eugeny/tabby/blob/1e5cfd1d4bd5a6dd23da60adc6ea1b6b5fd29e65/tabby-core/package.json#L31-L39
2. `tabby-core/dist/index.js` has `//# sourceMappingURL=index.js.map` at end but the source map file doesn't exist, causes a warning from `source-map-loader`
3. My project doesn't use Angular, so I have to import `@angular/compiler` before `tabby-web-container/dist/preload.js`, otherwise I have issue like https://stackoverflow.com/questions/70127946/the-injectable-platformlocation-needs-to-be-compiled-using-the-jit-compiler-b. I don't know Angular so I'm not sure what the best solution is.

**Reply by yume-chan** [reply]
`providerBlacklist` won't work if I minifies `tabby-settings/dist/index.js` (when I use Webpack assets import in prod mode it automatically does that).

**Reply by Eugeny** [reply]
* You're using the same loading scheme as tabby-web does, right? As a workaround, you can set `window.pluginModules` yourself after loading plugins: https://github.com/Eugeny/tabby-web/blob/e1b0d01ac320b9052c2d1ab3cc477d7c76689d31/frontend/src/terminal.ts#L53
* `@angular/compiler` is a part of `tabby-web-container/dist/bundle.js` bundle, and `tabby-web-container/dist/preload.js` doesn't require it (you should load `preload` first and `bundle` after that) 
* `providerBlacklist` relies on function names being preserved to work, it's possible to tell Webpack to minify everything except these: https://stackoverflow.com/questions/56482107/minify-while-keeping-function-names-with-webpack-and-babel
* I've updated peerDependencies, added sourcemaps and will publish packages in a minute

**Reply by yume-chan** [reply]
> You're using the same loading scheme as tabby-web does, right?

Yes, I have workarounded that.

> `@angular/compiler` is a part of `tabby-web-container/dist/bundle.js` bundle

The error comes from `require('@angular/cdk/clipboard')`, it's loaded before `require('@angular/compiler')`

https://github.com/Eugeny/tabby/blob/1b1445bf20a8c7b9bd8c3639e68c3bccccc0b0f8/web/polyfills.ts#L132-L134

**Comment by Eugeny** [comment]
Very nice project by the way!

**Reply by yume-chan** [reply]
Thank you! Tabby is the best terminal emulator on Web platform!
